src/sentry/snuba/discover.py

```diff
@@ -619,6 +630,21 @@ def timeseries_query(selected_columns, query, params, rollup, reference_event=No
                     conditions based on the provided reference.
     referrer (str|None) A referrer string to help locate the origin of this query.
     """
+    if top_events:
+        event_fields = top_events[0].fields
+        selected_columns += event_fields
+        top_events = {event.slug: find_reference_event(event, real=False) for event in top_events}
```

Comment:
It would be nice to do one snuba query to get all the event data, instead of one call for each event.